addons/l10n_ph/tests/test_bir_2307_generation.py

```diff
@@ -0,0 +1,88 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import io
+import xlrd
+import base64
+
+from odoo.addons.account_reports.tests.common import TestAccountReportsCommon
+
+from odoo import Command
+from odoo.tests import tagged
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestBIR2307Generation(TestAccountReportsCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref="ph"):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+
+        cls.company_data['company'].write({
+            'name': 'Test Company',
+            'street': '8 Super Street',
+            'city': 'Super City',
+            'zip': '8888',
+            'country_id': cls.env.ref('base.ph').id,
+            'vat': '123-456-789-123',
+        })
+        cls.partner_a.write({
+            'name': 'Test Partner',
+            'street': '9 Super Street',
+            'city': 'Super City',
+            'zip': '8888',
+            'country_id': cls.env.ref('base.ph').id,
+            'vat': '789-456-123-789',
+        })
+
+        # 1% Withholding Tax
+        vat_purchase_wc640 = cls.env.ref(f'account.{cls.company_data["company"].id}_l10n_ph_tax_purchase_wc640')
+
+        cls.invoice = cls.env['account.move'].create({
+            'move_type': 'in_invoice',
+            'invoice_date': '2020-01-15',
+            'partner_id': cls.partner_a.id,
+            'invoice_line_ids': [
+                Command.create({
+                    'name': 'Test line',
+                    'quantity': 1.0,
+                    'price_unit': 100,
+                    'tax_ids': vat_purchase_wc640,
+                })
+            ]
+        })
+        cls.invoice.action_post()
+
+    def _test_xls_file(self, file_content, expected_values):
+        """ Takes in the binary content of a xls file and a dict of expected values.
+        It will then parse the file in order to compare the values with the expected ones.
+        The expected values dict format is:
+        'row_number': ['cell_1_val', 'cell_2_val', ...]
+
+        :param file_content: The binary content of the xls file
+        :param expected_values: The dict of expected values
+        """
+        report_file = io.BytesIO(file_content)
+        xls = xlrd.open_workbook(file_contents=report_file.read())
+        sheet = xls.sheet_by_index(0)
+
+        for row, values in expected_values.items():
+            row_values = sheet.row_values(row)
+            for row_value, expected_value in zip(row_values, values):
+                self.assertEqual(row_value, expected_value)
```

Comment:
as there is only one test, I don't think making a helper here is needed. We can add it later if we ever add more tests.